android/app/src/main/java/updated/mysterium/vpn/ui/create/account/CreateAccountViewModel.kt

```diff
@@ -44,13 +45,15 @@ class CreateAccountViewModel(useCaseProvider: UseCaseProvider) : ViewModel() {
             _registrationError.postValue(exception as Exception)
         }
         viewModelScope.launch(handler) {
-            val nodeIdentity = connectionUseCase.getIdentity()
-            val identity = IdentityModel(
-                address = nodeIdentity.address,
-                channelAddress = nodeIdentity.channelAddress,
-                status = IdentityRegistrationStatus.parse(nodeIdentity.registrationStatus)
-            )
-            registerIntercomClient(identity.address)
+            if (!::newIdentity.isInitialized) {
```

Comment:
It's a bad idea to make `newIdentity` lateinit property. Better make it nullable and initialize initialy with null.
And then instead of checking initialization you will check is property null or not
```
if (newIdentity == null) {
    newIdentity  = ....
}

newIdentity. let { identity ->
    registerIntercomClient(identity )
}
```

Reply:
Fixed